### PR TITLE
no need in unsafe in topic name - deep copying

### PR DESCRIPTION
**Pull request by andrewmed**
I prefer https://github.com/ozontech/file.d/pull/54/files over this one

### CODE REVIEW

File: plugin/output/kafka/kafka.go [hunk 1]
```diff
@@ -162,11 +162,7 @@ func (p *Plugin) out(workerData *pipeline.WorkerData, batch *pipeline.Batch) {
 			data.messages[i] = &sarama.ProducerMessage{}
 		}
 		data.messages[i].Value = outBuf[start:]
-
-		// copy topic from json, to temporary out buffer to avoid event reusing issues
-		start = len(outBuf)
-		outBuf = append(outBuf, topic...)
-		data.messages[i].Topic = pipeline.ByteToStringUnsafe(outBuf[start:])
+		data.messages[i].Topic = pipeline.DeepCopyString(topic)
```
Comment (vano144) [inline plugin/output/kafka/kafka.go:165]: Let's just copy string only at 157 ? What do you think about it ?

File: pipeline/util.go [hunk 2]
```diff
@@ -8,6 +8,14 @@ import (
 	"unsafe"
 )
 
+// DeepCopyString create new string with underlying buffer
+func DeepCopyString(s string) string {
```
Comment (vano144) [inline pipeline/util.go:12]: https://go.googlesource.com/go/+/346fb7aaccda5cc059100502a02fe039ca0ea998/src/strings/clone.go ?

File: plugin/output/kafka/kafka.go (hunk 1 above)
Comment (andrewmed) [reply plugin/output/kafka/kafka.go]: Good point! 
Done

File: plugin/output/kafka/kafka.go [hunk 3]
```diff
@@ -154,19 +154,15 @@ func (p *Plugin) out(workerData *pipeline.WorkerData, batch *pipeline.Batch) {
 		if p.config.UseTopicField {
 			fieldValue := event.Root.Dig(p.config.TopicField).AsString()
 			if fieldValue != "" {
-				topic = fieldValue
+				topic = pipeline.CloneString(topic)
```
Comment (vano144) [inline plugin/output/kafka/kafka.go:157]: pipeline.CloneString(fieldValue)

Comment (andrewmed) [reply plugin/output/kafka/kafka.go]: 🤦‍♂️
fixed

**Comment by vano144** [comment]
LGTM

**Pull request merged**